client/common/directives/DateTimePicker/DateTimePicker.js

```diff
@@ -53,7 +53,7 @@ angular.module('dashboard.directive.DateTimePicker', [
           if (!scope.format) scope.format = scope.ngFormat;
           var dateValue = moment(elem.val(), scope.format);
           if (dateValue.isValid()) {
-            ngModel.$setViewValue(dateValue);
+            ngModel.$setViewValue(moment(dateValue).format(scope.format));
```

Comment:
dateValue looks like it is a moment object already, so you shouldn't have to create a new moment object again here.